File: Shortest Unsorted Continuous Subarray.py
class Solution:
     def findUnsortedSubarray(self, nums):
         """
         :type nums: List[int]
         :rtype: int
         """
         if len(nums)<2:
             return 0
         nums2=sorted(nums)
         s=0
         e=len(nums)-1
         while s<e:
             if nums2[s]==nums[s] and nums2[e]==nums[e]:
                 s+=1
                 e-=1
             elif nums2[s]==nums[s]:
                 s+=1
             elif nums2[e]==nums[e]:
                 e-=1
             else:
                 return 1+e-s
         return 0

File: Shortest Unsorted Continuous Subarray.cpp
class Solution {
 public:
     int findUnsortedSubarray(vector<int>& nums) {
         if(nums.size()<2)return 0;
         int l=-1,r=-1;
         vector<int> c(nums);
         sort(nums.begin(),nums.end());
         for(int i=0;i<nums.size();++i)
         {
             if(c[i]!=nums[i])
             {
                 l=i;
                 break;
             }
         }
         for(int i=nums.size()-1;i>=0;--i)
         {
             if(c[i]!=nums[i])
             {
                 r=i;
                 break;
             }
         }
         if(r==-1)return 0;
         return r-l+1;
         
     }
 };